src/pip/_internal/wheel.py

```diff
@@ -1202,6 +1198,21 @@ def build(
                         assert req.link.is_wheel
                         # extract the wheel into the dir
                         unpack_file(req.link.file_path, req.source_dir)
+                    else:
+                        # copy from cache to target durectory
+                        try:
+                            ensure_dir(self._wheel_dir)
+                            shutil.copy(
+                                os.path.join(output_dir, wheel_file),
+                                self._wheel_dir,
+                            )
+                        except OSError as e:
+                            logger.warning(
+                                "Building wheel for %s failed: %s",
```

Comment:
Yes, I chose to keep the same message that was used a few lines above in case creating the output directory fails. I'm not sure what is best: keeping a neutral message about building that failed (which from the user perspective is what happens - the wheel is not in the target directory), or being more specific about the failed copy operation (which could be seen as an implementation detail the user may not care about).

Reply:
Interestingly, in `_build_one_inside_env`, if the move from temp to the target directory fails, the exception is ignored so nothing is logged.

https://github.com/pypa/pip/blob/de736c661ef6d152d7a88569376ef12915b6ea16/src/pip/_internal/wheel.py#L975-L985

Reply:
I think I glossed over the fact that we're including the exception in the error message before. Does this include the actual directory? If so then this is OK as-is for me, the important thing here is being able determine what actually failed when given user logs.

Reply:
Yes, here is a typical error message. With master, the WARNING is absent, you only get `Failed to build pyyaml`, due to the `except Exception: pass` in `_build_one_...`.

```console
$ pip wheel pyyaml==5.1.2 --wheel-dir=/tmp/readonlydir                                                        1  tempenv-4af21963d2a 
Collecting pyyaml==5.1.2
  Using cached PyYAML-5.1.2.tar.gz
Building wheels for collected packages: pyyaml
  Building wheel for pyyaml (setup.py) ... done
  Created wheel for pyyaml: filename=PyYAML-5.1.2-cp37-cp37m-linux_x86_64.whl size=44103 sha256=007225a64a0081b8526bad271dfe406c868c85e110b340d4527505ed04f3990c
  Stored in directory: /home/me/.cache/pip/wheels/d9/45/dd/65f0b38450c47cf7e5312883deb97d065e030c5cca0a365030
  WARNING: Building wheel for pyyaml failed: [Errno 13] Permission denied: '/tmp/readonlydir/PyYAML-5.1.2-cp37-cp37m-linux_x86_64.whl'
Failed to build pyyaml
ERROR: Failed to build one or more wheels
```